packages/data-broker/src/services/dhis/pullers/AnalyticsPuller.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import groupBy from 'lodash.groupby';
+import keyBy from 'lodash.keyby';
+
+import {
+  buildAnalyticsFromDhisAnalytics,
+  buildAnalyticsFromDhisEventAnalytics,
+  buildAnalyticsFromEvents,
+} from '../buildAnalytics';
+
+export class AnalyticsPuller {
+  constructor(dataSourceModel, translator, dataElementsMetadataPuller) {
+    this.dataSourceModel = dataSourceModel;
+    this.translator = translator;
+    this.dataElementsMetadataPuller = dataElementsMetadataPuller;
+  }
+
+  groupDataSourcesByDhisDataType = dataSources =>
+    groupBy(
+      dataSources,
+      d => d.config?.dhisDataType || this.dataSourceModel.getDhisDataTypes().DATA_ELEMENT,
+    );
+
+  groupIndicatorDataSourcesByPeriodType = dataSources => {
```

Comment:
I've also put a comment in buildAggregatedDataValueAnalyticsQueries which I think is correct, read that first and let me know if I'm wrong. But I think this is not just an indicator problem. Technically, someone could come along and fetch data element values through our aggregated method... they would be crazy to do so but they could. Do we want to rename the method here to remove the mention of Indicator? Happy to leave as is, just a thought.